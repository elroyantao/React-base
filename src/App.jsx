import React, { Component } from 'react'

export default class App extends Component {
  render() {
    return (
      <div className="App">
          This is a react minimalist boilerplate (Works)
      </div>
    )
  }
}