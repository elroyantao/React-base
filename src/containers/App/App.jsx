import React, { Component } from 'react'
import { connect } from 'react-redux'

import { incrementCount } from '../../actions/homeActions'

class App extends Component {
  render() {
    const { count, incrementCount } = this.props
    return (
      <div className="App">
          <div>This is a react minimalist boilerplate</div>
          <br />
          <div>count : {count}</div>
          <button onClick={incrementCount}>Click to Increment Count</button>
      </div>
    )
  }
}

export default connect((state) => ({
  count: state.home.count
}), { incrementCount })(App)
