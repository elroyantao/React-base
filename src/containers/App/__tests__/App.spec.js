import TestComponentHelper from '../../../lib/test-helper'
import App from '../App'

describe('<App />', () => {
  const renderComponent = TestComponentHelper(App)
  describe('@renders', () => {
    it('in default state', () => {
      const { getTree } = renderComponent()
      expect(getTree()).toMatchSnapshot()
    })
  })
})
