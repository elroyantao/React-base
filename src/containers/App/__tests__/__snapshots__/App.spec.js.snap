// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> @renders in default state 1`] = `
<div
  className="App"
>
  <div>
    This is a react minimalist boilerplate
  </div>
  <br />
  <div>
    count : 
  </div>
  <button>
    Click to Increment Count
  </button>
</div>
`;
